src/EmailAddress.php

```diff
@@ -491,15 +491,17 @@ public function isValid($value)
         }
 
         $length  = true;
-        $this->setValue($this->idnToUtf8($value));
 
         // Split email address up and disallow '..'
-        if (!$this->splitEmailParts($this->getValue())) {
+        if (!$this->splitEmailParts($value)) {
             $this->error(self::INVALID_FORMAT);
             return false;
         }
 
-        if ((strlen($this->localPart) > 64) || (strlen($this->hostname) > 255)) {
+        $this->hostname = $this->idnToUtf8($this->hostname);
+
+        // https://tools.ietf.org/html/rfc5322#section-2.1.1
+        if ((strlen($this->localPart) + strlen($this->hostname)) > 994) {
```

Comment:
I actually subtracted 4 from 998 to account for `To:` and `@`, but did not document it properly.
Thanks for noticing.